heracles-resources/backend/src/main/java/io/uhndata/cards/heracles/internal/migrators/Heracles8PAMigration.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.uhndata.cards.heracles.internal.migrators;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import javax.jcr.Node;
+import javax.jcr.NodeIterator;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+import javax.jcr.query.Query;
+import javax.jcr.version.VersionManager;
+
+import org.osgi.framework.Version;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.uhndata.cards.forms.api.FormUtils;
+import io.uhndata.cards.migrators.spi.DataMigrator;
+
+@Component(immediate = true)
+public class Heracles8PAMigration implements DataMigrator
+{
+    private static final String PA_REASON_QUESTION_PATH = "/Questionnaires/Physical Assessments/pa_reason";
+    private static final String PA_QUESTIONNAIRE_PATH = "/Questionnaires/Physical Assessments";
+    private static final String VISIT_NUMBER_SECTION_PATH = "/Questionnaires/Physical Assessments/section_visit_number";
+    private static final String OLD_VISIT_NUMBER_QUESTION_PATH = "/Questionnaires/Physical Assessments/visit_number";
+    private static final String NEW_VISIT_NUMBER_QUESTION_PATH
```

Comment:
I would reorder these as:

```suggestion
    private static final String PA_QUESTIONNAIRE_PATH = "/Questionnaires/Physical Assessments";
    private static final String PA_REASON_QUESTION_PATH = "/Questionnaires/Physical Assessments/pa_reason";
    private static final String OLD_VISIT_NUMBER_QUESTION_PATH = "/Questionnaires/Physical Assessments/visit_number";
    private static final String NEW_VISIT_NUMBER_SECTION_PATH = "/Questionnaires/Physical Assessments/section_visit_number";
    private static final String NEW_VISIT_NUMBER_QUESTION_PATH
```

Note that the diff above also renames the section path variable, so it shouldn't be applied as-is without further code changes, and the line may also be longer than the allowed limit.